### ISSUE TITLE
Can you explain model for training ?

### ISSUE THREAD

**Issue by ThorPham** [kurapan/EAST#17]
I read paper "EAST: An Efficient and Accurate Scene Text Detector" but i don't understand some point.
In your code 
`model = Model(inputs=[input_image, overly_small_text_region_training_mask, text_region_boundary_training_mask, target_score_map], outputs=[pred_score_map, pred_geo_map])`

What is overly_small_text_region_training_mask, text_region_boundary_training_mask, target_score_map ? Where is it come and how to calculation .

**Comment by kurapan** [comment]
The original code contains a training mask which masks out regions with small text. I don't think it masks out the boundaries of text regions.
The point of `overly_small_text_region_training_mask` is to mask out text instances which are smaller than some threshold so we don't expect the model to find them, but we don't want to penalize the model in case it is able to detect it. That means we want to ignore that region when calculating the loss of the score map (binary segmentation part).
Similarly, as the masks for the segmentation are shrinked as described in the paper, the boundaries of text regions are not included in the segmentation ground truth. The boundaries of the text are then ignored when calculating the loss, which means that we don't care if they are classified as positive or negative. It shouldn't make much difference if the boundaries are masked out or not, though.
As for `target_score_map`, that's the regular segmentation ground truth as described in the paper.

I don't really understand what you mean by how to calculate it.
The loss functions are defined in the code and used to train the model. The evaluation script then can be used to detect text regions in test images. And as regards calculation of precision, recall, and F1 score, you can use the output of the evaluation script and use the official ICDAR evaluation code for ICDAR datasets.

**Comment by ThorPham** [comment]
@kurapan Thank so much.

**Comment by fzyzcjy** [comment]
+1 for that question. Thanks!
